src/search/srpso.c

```
@@ -0,0 +1,239 @@
+#include "search.h"
+#include "search_optimizer.h"
+#include "sr_alloc.h"
+#include "sr_rng.h"
+
+static int sr_pso_default_swarm_size(int ndim)
+{
+  const int base = 5 * ndim;
+  return (base < 10) ? 10 : base;
+}
+
+void sr_pso_cfg_init(sr_pso_cfg *cfg, int ndim, real dpos)
```

Clang (14.0.6)

[warning] parameter 'cfg' is unused

(misc-unused-parameters)